geode-core/src/distributedTest/java/org/apache/geode/management/internal/cli/commands/AlterQueryServiceCommandWithSecurityDUnitTest.java

```diff
@@ -0,0 +1,256 @@
+        i -> servers.add(i, cluster.startServerVM(i + 1, s -> s.withConnectionToLocator(locatorPort)
+            .withCredential("clusterManage", "clusterManage"))));
+    gfsh.connectAndVerify(locator);
+  }
+
+  // This test verifies that the cluster starts with the default method authorizer configured, then
+  // changes the currently configured authorizer to each of the out-of-the-box authorizers.
+  @Test
+  public void alterQueryServiceCommandUpdatesMethodAuthorizer() {
+    Class<RestrictedMethodAuthorizer> restrictedAuthorizerClass = RestrictedMethodAuthorizer.class;
+    Class<UnrestrictedMethodAuthorizer> unrestrictedAuthorizerClass =
+        UnrestrictedMethodAuthorizer.class;
+    Class<JavaBeanAccessorMethodAuthorizer> javaAccessorAuthorizerClass =
+        JavaBeanAccessorMethodAuthorizer.class;
+    Class<RegExMethodAuthorizer> regExAuthorizerClass = RegExMethodAuthorizer.class;
+
+    verifyCurrentAuthorizerClass(DEFAULT_AUTHORIZER_CLASS);
+
+    String authorizerName = unrestrictedAuthorizerClass.getName();
+    gfsh.executeAndAssertThat(COMMAND_NAME + " --" + METHOD_AUTHORIZER_NAME + "=" + authorizerName)
+        .statusIsSuccess().hasTableSection().hasRowSize(serversToStart);
+
+    verifyCurrentAuthorizerClass(unrestrictedAuthorizerClass);
+
+    authorizerName = javaAccessorAuthorizerClass.getName();
+    gfsh.executeAndAssertThat(COMMAND_NAME + " --" + METHOD_AUTHORIZER_NAME + "=" + authorizerName
+        + " --" + AUTHORIZER_PARAMETERS + "=" + JAVA_BEAN_AUTHORIZER_PARAMS).statusIsSuccess()
+        .hasTableSection().hasRowSize(serversToStart);
+
+    verifyCurrentAuthorizerClass(javaAccessorAuthorizerClass);
+    servers.forEach(server -> server.invoke(() -> {
+      JavaBeanAccessorMethodAuthorizer methodAuthorizer = (JavaBeanAccessorMethodAuthorizer) Objects
+          .requireNonNull(ClusterStartupRule.getCache()).getService(QueryConfigurationService.class)
+          .getMethodAuthorizer();
+
+      Set<String> expectedParameters =
+          new HashSet<>(Arrays.asList(JAVA_BEAN_AUTHORIZER_PARAMS.split(",")));
+      assertThat(methodAuthorizer.getAllowedPackages()).isEqualTo(expectedParameters);
+    }));
+
+    authorizerName = regExAuthorizerClass.getName();
+    gfsh.executeAndAssertThat(COMMAND_NAME + " --" + METHOD_AUTHORIZER_NAME + "=" + authorizerName
+        + " --" + SINGLE_AUTHORIZER_PARAMETER + "=" + REGEX_AUTHORIZER_PARAMETER).statusIsSuccess()
+        .hasTableSection().hasRowSize(serversToStart);
+
+    verifyCurrentAuthorizerClass(regExAuthorizerClass);
+    servers.forEach(server -> server.invoke(() -> {
+      RegExMethodAuthorizer methodAuthorizer = (RegExMethodAuthorizer) Objects
+          .requireNonNull(ClusterStartupRule.getCache()).getService(QueryConfigurationService.class)
+          .getMethodAuthorizer();
+
+      assertThat(methodAuthorizer.getAllowedPatterns().size()).isEqualTo(1);
+      assertThat(methodAuthorizer.getAllowedPatterns()).containsOnly(REGEX_AUTHORIZER_PARAMETER);
```

Comment:
I feel at this point, this could be broken into a seperate test as we are now testing that the pattern and the Regex specific authorizor

Reply:
These lines are just checking that the value for `authorizer-parameter` specified via the gfsh command is correctly set on the method authorizer that is present on the servers. It's not testing the behaviour or functionality of that method authorizer at all.

Reply:
A new test has been added to specifically verify that parameters are correctly set